calico/felix/actor.py

```diff
@@ -226,12 +237,30 @@ def _step(self):
                 actor_storage.msg_uuid = msg.uuid
                 actor_storage.msg_name = msg.name
                 try:
+
+                    # Keeps track of how many times we retry a message.
+                    try:
+                        # Increment the counter if it already exists
+                        msg_retries[msg.uuid] += 1
+                    except KeyError:
+                        # Else set it up
+                        msg_retries[msg.uuid] = 1
```

Comment:
Rather than doing this yourself, you might want to look at the [`collections` module](https://docs.python.org/2/library/collections.html). The `defaultdict` class provides essentially the same functionality as what you've written here, but it's a bit cleaner.